src/argparse_utils.h

```diff
@@ -28,10 +28,12 @@
 bool parse_common_args(const std::string& program,
                        const cxxopts::Options& opts,
                        const cxxopts::ParseResult& result,
-                       boost::property_tree::ptree& conf,
+                       std::optional<boost::property_tree::ptree> conf,
                        const std::string& log,
                        const bool use_threads = false,
                        std::function<void()> extra_help = nullptr) {
+  bool needs_conf = conf.has_value();
```

Comment:
for valhalla_service we need to set up custom logging when using one-shot mode (where we print to `stderr`), also we don't support `--config` etc args which renders much of this function useless. it still sets up the help & version stuff though